File: SourceCode/C/day052.c
/* Author: Ahmed A. M. ALOMAR
Date: February 21 - 2020
Description:
This program is an example of  function return 
This program will add two to the number provided and print out the results. */


#include <stdio.h>

int adding_two(int x);


int main (void)
{
  int result = 0;

  adding_two(0);

  adding_two(10);

  adding_two(20);

  return 0;
}


int adding_two(int x)
{
  int result = x += 2;

  printf("The first result: %d\n", result);
  printf("The second result: %d\n", result);
  printf("The third result: %d\n\n", result);

  return result;
}

File: SourceCode/Python/day052.py
# ------------------------------------ #
# Author: Ahmed A. M. ALOMAR
# Date: June 1 - 2020
#
# Description:
# * Example of Heigher Order Function
# ------------------------------------ #


# a small function
def add(x, y):
  return x + y

# Another small function
def subtract(x, y):
  return x - y


# The Heigher order function
def calculate(operation, x, y):
  return operation(x, y)


print(calculate(add, 5, 8))
